src/audio/volume/volume.c

```diff
@@ -755,10 +948,14 @@ static int volume_copy(struct comp_dev *dev)
 		 c.source_bytes, c.sink_bytes);
 
 	while (c.frames) {
+#if CONFIG_IPC_MAJOR_4
+		volume_update_current_vol_ipc4(cd);
+#endif
+
```

Comment:
See above for init. We can change this to always call ```volume_update_current_peak()``` where it can be defined as
```c
static inline void volume_update_current_peak(cd) {}
```
for IPC3 but as above for IPC4.